### ISSUE TITLE
Some questions about training details

### ISSUE THREAD

**Issue by jz-exwzd** [neuralchen/SimSwap#159]
Hi @NNNNAI @neuralchen @AceSix @bfirsh @Arthurzhangsheng,

  Since the training codes are still not yet released, I am trying to write it myself. It would be nice of you if you could at least answer some of my questions about the training details.

  1) I noticed that the weights for different losses (lambdas) are different from the paper. In the paper, it is lambda_id=10, lambda_recon=10, lambda_gp=10e-5, lambda_FM=10. However, in the opt.txt text file which contains the option information about your training, the lambdas are lambda_id=20, lambda_recon=10, lambda_gp=10, lambda_FM=10. Which set of lambdas should I use?

  2) There is an extra term loss_vgg which uses VGG19 network I suppose. I did not see this component in the paper. Should I include this loss? If yes, what should the entry be for the VGG19 network to calculate this loss? Also, what is the lambda for this loss?

  3) I noticed that the reconstruction loss is always applied between the generated output and the attribute image, which is different from what is described in the paper, where you mentioned that this loss is set to 0 when the attribute image and ID image are different. So, which one is correct?

  4) I equally noticed that the attribute image and the generated output are downsampled and fed into the discriminators. This is not described in the paper. Could you explain the rational of this? 

5) Lastly, I would like to know how long you trained the model. In your paper, it is written that you trained the network for more than 500 epoch. Based on your opt.txt, 1 epoch consists of niter=10000 iterations. So, the model is trained for 5000 000 steps, am I right? The learning rate is also decayed every epoch if I am not mistaken.

  I am sorry for all the questions, but since there are no training codes, it would be nice if you could at least provide some supports for writing the training code of the model.

Hope to hear from you soon.

Thank you.

**Comment by NNNNAI** [comment]
Hi, I will reply your question as follow.
1: the default value of  argument is not same as the paper and also not same as current version, you can use this in command line --beta1 0.0 --lambda_id 10.0 to get the same effect.
2: We dont not use VGG19.
3:For example.in first step the target image and source image is came from the same person. So the the generated output and source image (which you called the attribute image) is same id, which is make sence to caluate rec loss. and in the second step we will make target image and source image is came from the different person.And we will set the weight to 0.
4:We did mention this trick in paper.This is a multisalce discriminators.
5: It takes me 5 days to train on 2080ti with bs 8

**Comment by jz-exwzd** [comment]
Hi @NNNNAI ,

  Thanks for replying. I think I understand the answer for my questions from 2. to 5.

  About the lambdas,  can you let me know what should the values of the following weights be if I want to retrain the model myself?
lambda_id 
lambda_recon 
lambda_gp
lambda_FM

  Especially, should the lambda_gp be 10 or 10e-5?

  Thank you.

**Comment by NNNNAI** [comment]
lambda_gp:10e-5 , lambda_recon: 10, lambda_FM:10, lambda_id:10

**Comment by jz-exwzd** [comment]
Thanks!

**Comment by BhargaviAndra** [comment]
is the learning rate 0.0002 as mentioned in the options?
